deprecated/frontend/schema/class-schema-author.php

```diff
@@ -33,25 +53,35 @@ public function __construct() {
 	 * @return bool
 	 */
 	public function is_needed() {
-		_deprecated_function( __METHOD__, 'WPSEO xx.x' );
+		_deprecated_function( __METHOD__, 'WPSEO xx.x', 'Yoast\WP\SEO\Generators\Schema\Author::is_needed' );
+
+		$context = $this->memoizer->for_current_page();
 
-		return false;
+		return $this->author->is_needed( $context );
 	}
 
 	/**
 	 * Returns Person Schema data.
 	 *
+	 * @codeCoverageIgnore
+	 * @deprecated xx.x
+	 *
 	 * @return bool|array Person data on success, false on failure.
 	 */
 	public function generate() {
-		_deprecated_function( __METHOD__, 'WPSEO xx.x' );
+		_deprecated_function( __METHOD__, 'WPSEO xx.x', 'Yoast\WP\SEO\Generators\Schema\Author::generate' );
 
-		return array();
+		$context = $this->memoizer->for_current_page();
+
+		return $this->author->generate( $context );
 	}
 
 	/**
 	 * Gets the Schema type we use for this class.
 	 *
+	 * @codeCoverageIgnore
+	 * @deprecated xx.x
+	 *
 	 * @return string[] The schema type.
 	 */
 	public static function get_type() {
```

Comment:
Although the original `get_type` method did not work and errors (it [tries to return a non-static variable 😛](https://github.com/Yoast/wordpress-seo/blob/0f95ac82f83b9fc63d9422fb2f7388f314e7a58c/frontend/schema/class-schema-author.php#L143) ) it would be nice to fix it here by returning `[ 'Person' ]` instead of an empty array.